sway-lib-std/src/context/call_frames.sw

```diff
@@ -43,19 +49,13 @@ pub fn code_size() -> u64 {
 }
 
 /// Get the first parameter from the current call frame.
-pub fn first_param() -> u64 {
-    asm(size, ptr, offset: 584) {
-        add size fp offset;
-        size: u64
-    }
+pub fn first_param<T>() -> T {
```

Comment:
The first parameter is always a `u64` (technically `u32`), so don't think this needs to be generic.